From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Greg Price <greg@zulip.com>
Date: Tue, 24 May 2022 16:58:12 -0700
Subject: [tsflower] rnsac: Fix $ReadOnlyArray reference to literally use that
 name

Otherwise get an error on trying to pass an array literal here;
even if we mark the property read-only.

I suspect the Flow special handling for array literals is looking
specially for `$ReadOnlyArray`.  That means TsFlower should rewrite
TS references to `ReadonlyArray` to simply use the name
`$ReadOnlyArray` directly, instead of using its general "subst"
mechanism.
---
 .../lib/typescript/SafeArea.types.js.flow                      | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git types/react-native-safe-area-context/lib/typescript/SafeArea.types.js.flow types/react-native-safe-area-context/lib/typescript/SafeArea.types.js.flow
index c1d8cbafe..88ba6cc09 100644
--- types/react-native-safe-area-context/lib/typescript/SafeArea.types.js.flow
+++ types/react-native-safe-area-context/lib/typescript/SafeArea.types.js.flow
@@ -7,7 +7,6 @@ import type {
 } from 'tsflower/subst/react-native';
 
 import type { ReactNode as $tsflower_subst$React$ReactNode } from 'tsflower/subst/react';
-import type { ReadonlyArray } from 'tsflower/subst/lib';
 import * as React from 'react';
 import 'react-native';
 export type Edge = 'top' | 'right' | 'bottom' | 'left';
@@ -42,5 +41,5 @@ export interface NativeSafeAreaProviderProps extends $tsflower_subst$RN$ViewProp
 export interface NativeSafeAreaViewProps extends $tsflower_subst$RN$ViewProps {
   children?: $tsflower_subst$React$ReactNode;
   mode?: 'padding' | 'margin';
-  edges?: ReadonlyArray<Edge>;
+  edges?: $ReadOnlyArray<Edge>;
 }
-- 
2.32.0

